packages/superset-ui-core/src/query/normalizeOrderBy.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import isEmpty from 'lodash/isEmpty';
+import isBoolean from 'lodash/isBoolean';
+
+import { QueryObject } from './types';
+
+export default function normalizeOrderBy(queryObject: QueryObject): QueryObject {
+  if (Array.isArray(queryObject.orderby) && queryObject.orderby.length > 0) {
+    // ensure a valid orderby clause
+    const orderbyClause = queryObject.orderby[0];
+    if (
+      Array.isArray(orderbyClause) &&
+      orderbyClause.length === 2 &&
+      !isEmpty(orderbyClause[0]) &&
+      isBoolean(orderbyClause[1])
+    ) {
+      return queryObject;
+    }
+  }
+
+  // ensure that remove invalid orderby clause
+  const cloneQueryObject = { ...queryObject };
```

Comment:
I think `const cloneQueryObject = omit(queryObject, ['timeseries_limit_metric', 'order_desc', 'orderby'])` can be more concise. It returns a new object.

Reply:
the typescript `omit` seem that return a type.

https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys

Reply:
here we can use lodash omit

Reply:
Oh, thanks! Follow package core's pattern, and try to use a more "native" approach at here.